### PR TITLE
[Libraries] Remove unused functions from Utility and update cleanup comments

### PR DESCRIPTION
**Pull request by johnsaigle**
- [x] Delete functions marked as unused in the function comments (and confirmed by a Github search).

- [x] Update function comments to reflect some locations where they are used

### CODE REVIEW

**Comment by driusan** [comment]
This needs to go to 18.0. There's no way to determine that they're unused by any project overrides or instruments in the wild.

**Comment by johnsaigle** [comment]
Note: changes made by @AnyhowStep in #2889 have now been merged into this PR. The other PR will be closed.

**Comment by johnsaigle** [comment]
@ridz1208 This PR is an update of #2889 which was opened by @AnyhowStep. You said his PR should go to 17.2.

Do you think should be in 17.2 or 18.0? My vote is 17.2 but @driusan said 18.0. Do you have a preference?

File: php/libraries/Utility.class.inc [hunk 1]
```diff
@@ -150,36 +150,6 @@ class Utility
...
-    /**
-     * Returns a list of existing ethnicities of all candidates
-     *
-     * @return  array       an associative array("ethnicity" => "ethnicity")
-     * @access  public
-     * @static
-     * @note    This is no longer used and should be removed
-     * @cleanup
-     */
-    static function getListOfSubjectEthnicities()
```
Comment (ridz1208) [inline php/libraries/Utility.class.inc]: confirmed not being used

File: php/libraries/Utility.class.inc [hunk 2]
```diff
@@ -574,73 +555,6 @@ class Utility
...
-     * @param string $string    unknown
-     *
-     * @return unknown
-     *
-     * @note    Don't know what the above function comment is supposed to mean,
-     *       so just preserving it for now. Appears to only be used
-     *       by NDB_Form_candidate_parameters and if so, should be moved there.
-     * @cleanup
-     */
-    static function explodeEncapsulated($seperator, $delimiter, $string)
```
Comment (ridz1208) [inline php/libraries/Utility.class.inc]: used in candidate_parameters on IBIS

@sruthymathew123 if you can get rid of it or move it to the class it is used in, then when you confirm we'll be able to remove safely

File: php/libraries/Utility.class.inc [hunk 3]
```diff
@@ -574,73 +555,6 @@ class Utility
...
-     * why this exists instead of using built-in PHP functions.
-     * to be used.
-     *
-     * From php.net
-     *
-     * @return  The parameters passed all merged together, with keys preserved?
-     * @note    Function comment written by Dave, not the author of this function.
-     * @note    This function does not appear to be used anymore.
-     * @cleanup
-     */
```
Comment (ridz1208) [inline php/libraries/Utility.class.inc]: confirmed not used

File: php/libraries/Utility.class.inc [hunk 4]
```diff
@@ -669,62 +587,6 @@ class Utility
...
-     *
-     *
-     * If this doesn't make sense, see note.
-     *
-     * @param array $Arr A row from an associative array based by array_map
-     *
-     * @return  The name from the given array row.
-     * @note    Function should be removed and code using it cleaned up.
-     * @cleanup
-     */
```
Comment (ridz1208) [inline php/libraries/Utility.class.inc]: confirmed not used

File: php/libraries/Utility.class.inc [hunk 5]
```diff
@@ -669,62 +587,6 @@ class Utility
...
```
Comment (ridz1208) [inline php/libraries/Utility.class.inc]: confirmed not used

File: php/libraries/Utility.class.inc [hunk 6]
```diff
@@ -849,33 +711,14 @@ class Utility
...
-     * overflow into a new table. This information comes from the config.xml.
-     *
-     * @return integer Column threshold from config file
-     * @note   Function should be either removed or moved to NDB_Config
-     * @note   Function comment written by Dave, who can't remember if he's
-     *       the author of this function.
-     *
-     * @cleanup
-     */
-    static function getColumnThresholdCount()
```
Comment (ridz1208) [inline php/libraries/Utility.class.inc]: confirmed not used

File: php/libraries/Utility.class.inc [hunk 7]
```diff
@@ -895,34 +738,6 @@ class Utility
...
-     *                          which visit labels exist in.
-     *
-     * @return Array of the form array(0 => array('Visit_label' => VL))
-     *         where VL is the value returned from the database.
-     *
-     * @note    This should be moved out of the Utility class into whatever
-     *       module uses it. (Data team helper?)
-     * @note    Function comment written by Dave, not the author of this function.
-     * @cleanup
-     */
```
Comment (ridz1208) [inline php/libraries/Utility.class.inc]: IBIS redefines this function in its ProjectUtility class but neither functions are used anywhere in the code

@sruthymathew123 potential clean-up for you !

File: php/libraries/Utility.class.inc [hunk 8]
```diff
@@ -983,37 +798,6 @@ class Utility
...
-    /**
-     * Get all the sourcefrom instruments from parameter_type (which
-     * exist in the data_integrity flag table)
-     *
-     * @return  Associative array of the form array('Full Name' => 'Full Name')
-     * @note    This should be moved out of the Utility class into whatever
-     *       module uses it. (Data team helper?)
-     * @note    Function comment written by Dave, not the author of this function.
-     * @cleanup
-     */
```
Comment (ridz1208) [inline php/libraries/Utility.class.inc]: same...
IBIS redefines this function in its ProjectUtility class but neither functions are used anywhere in the code

@sruthymathew123 potential clean-up for you !

File: php/libraries/Utility.class.inc [hunk 9]
```diff
@@ -660,6 +574,10 @@ class Utility
      *                      to null.
      *
      * @return array The same array passed in, after modifications.
```
Comment (ridz1208) [inline php/libraries/Utility.class.inc]: note should be removed

File: php/libraries/Utility.class.inc [hunk 10]
```diff
@@ -565,6 +544,8 @@ class Utility
      * @return  array If $var is an array, var, otherwise an array containing $var
      * @cleanup This should be removed and all uses converted to toArray
      *          (or vice versa, but toArray seems to be more common in the code)
```
Comment (ridz1208) [inline php/libraries/Utility.class.inc]: note should be removed

File: php/libraries/Utility.class.inc [hunk 11]
```diff
@@ -546,6 +523,8 @@ class Utility
      * @param mixed $var The variable to be converted to an array.
      *
      * @return array If $var is an array, var, otherwise an array containing $var
```
Comment (ridz1208) [inline php/libraries/Utility.class.inc]: note should be removed

**Comment by ridz1208** [comment]
@zaliqarosli let's talk about this one

File: php/libraries/Utility.class.inc [hunk 12]
```diff
@@ -849,33 +697,14 @@ class Utility
...
-    }
-
     /**
      * Checks to see if a table contains a specified column
      *
      * @param string $table_name The table to check for a column.
      * @param string $column     The column name to check the table for.
      *
      * @return boolean true if the table has at least one NULL in the column
+     * @note   This should really be in the Database.class.inc file... -anyhowstep
```
Comment (zaliqarosli) [inline php/libraries/Utility.class.inc]: I left this note in .. @ridz1208 should it be moved to Database?

**Comment by zaliqarosli** [comment]
This branch seems to be well out of sync, it doesn't generate a front-end for me. Need rebasing? @driusan @ridz1208

File: php/libraries/Utility.class.inc (hunk 12 above)
Comment (driusan) [reply php/libraries/Utility.class.inc]: probably, but not as part of this PR, and the note is still pointless to have. If anything, there should be a redmine ticket for it (or just a PR that does it and deprecates this)

File: php/libraries/Utility.class.inc (hunk 8 above)
Comment (driusan) [reply php/libraries/Utility.class.inc]: I think this was used by the really old "quat table" DQT. It's unused, and IBIS ProjectUtility probably just copied the Utility to start with.

File: php/libraries/Utility.class.inc [hunk 13]
```diff
@@ -1123,7 +893,10 @@ class Utility
      *               array(0 => array('Test_name_display' => $TestName))?
      *
      * @note    Function comment written by Dave, not the author of this function.
-     *       The return value format seems weird. Where is this used?
+     *       The return value format seems weird. Should possibly be refactored.
+     * @note    Used in:
+     *      - data_team_helper/php/NDB_Form_data_team_helper.class.inc
+     *      - data_team_helper/ajax/GetInstruments.php
```
Comment (driusan) [inline php/libraries/Utility.class.inc]: A redmine ticket saying to move this to the data_team_helper and verify that that's the only place that it's used is better than a comment saying where it's used (which would need to be maintained, and may become outdated..)

File: php/libraries/Utility.class.inc (hunk 12 above)
Comment (ridz1208) [reply php/libraries/Utility.class.inc]: I'm not against the notes, i tend to look at code more then redmine myself

**Comment by ridz1208** [comment]
@zaliqarosli I will let you attempt a rebase on your local repo. if you encounter any problems let me know

**Comment by zaliqarosli** [comment]
@ridz1208 please re-review. thanks!

**Comment by ridz1208** [comment]
@driusan I think this should be merge early if you agree

**Pull request merged**